src/libsystemd/sd-journal/journal-file.c

```diff
@@ -3550,7 +3550,7 @@ int journal_file_move_to_entry_by_monotonic_for_data(
 
         /* And now, continue seeking until we find an entry that exists in both bisection arrays. */
```

Comment:
Not exactly related to your changes, but I'm super confused by this comment, why do we need to find an entry that exists in both bisection arrays? Can you expand this comment a bit so I understand what's going on in this function? That would help me to review the changes.